File: StringRotaion/StringRotation.js
/**
 * Created by ken on 12/29/14.
 */
function stringRotation(str1, str2){
    if(str1.length == str2.length){
        if(str1.concat(str1).indexOf(str2) > 0){
            return true;
        }
    }
    return false;
}

console.log(stringRotation("california", "niacalifor"));

File: StringRotaion/StringRotation.java
public class StringRotation {
	public static void main(String[] args) {
		System.out.println(checkRotation("california", "niacalifor"));
	}

	public static boolean checkRotation(String s1, String s2){
		if(s1.length() == s2.length()){
			return s1.concat(s1).toLowerCase().contains(s2.toLowerCase());
		}
		return false;
	}
}